src/cdk-experimental/selection/selection.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, EventEmitter, OnInit, Input, ElementRef} from '@angular/core';
+import {SelectionChange, SelectionModel, compareFn} from '@angular/cdk/collections';
+import {coerceBooleanProperty, coerceNumberProperty} from '@angular/cdk/coercion';
+
+
+@Directive({
+  selector: '[cdkSelection]'
+})
+export class CdkSelection<T> implements OnInit {
+
+  /** List of all active selections. */
+  @Input('cdkSelection') selections: T[] = [];
```

Comment:
re: set - I suppose it could be but from a public API perspective that feels odd. I totally see the value of doing that internally though. Also, the API of the selection model class is a array so since this is more just directive bindings for that I was trying to honor that api. Let me know your thoughts.